sdk/src/akkaserverless.ts

```diff
@@ -0,0 +1,439 @@
+    if (bindings) {
+      if (bindings.address) {
+        this.address = bindings.address;
+      }
+      if (bindings.port) {
+        this.port = bindings.port;
+      }
+    }
+
+    const allComponentsMap: any = {};
+    this.components.forEach((component: Component) => {
+      allComponentsMap[component.serviceName ?? 'undefined'] =
+        component.service;
+    });
+
+    const componentTypes: any = {};
+    this.components.forEach((component: Component) => {
+      if (component.register) {
+        const componentServices = component.register(allComponentsMap);
+        componentTypes[componentServices.componentType()] = componentServices;
+      }
+    });
+
+    Object.values(componentTypes).forEach((services: any) => {
+      services.register(this.server);
+    });
+
+    const discoveryServer = this.getDiscoveryServer();
+
+    this.server.addService(discovery_grpc.DiscoveryService, discoveryServer);
+
+    return new Promise((resolve, reject) => {
+      this.server.bindAsync(
+        `${this.address}:${this.port}`,
+        grpc.ServerCredentials.createInsecure(),
+        (err, port) => {
+          if (err) {
+            console.error(`Server error: ${err.message}`);
+            reject(err);
+          } else {
+            console.log(`Server bound on port: ${port}`);
+            this.server.start();
+            this.afterStart(port);
+            resolve(port);
+          }
+        },
+      );
+    });
+  }
+
+  docLinkFor(code: string) {
+    return this.docLink.getLink(code);
+  }
+
+  formatSource(location: discovery.UserFunctionError.SourceLocation) {
+    return new SourceFormatter(location).getLocationString(this.components);
+  }
+
+  getDiscoveryServer() {
+    const that = this;
+    const discoveryServer: discovery_grpc.IDiscoveryServer = {
+      discover(
+        call: grpc.ServerUnaryCall<discovery.ProxyInfo, discovery.Spec>,
+        callback: grpc.sendUnaryData<discovery.Spec>,
+      ) {
+        console.debug('Received request');
+        const result = that.discoveryLogic(call.request);
+        callback(null, result);
+      },
+      reportError(
+        call: grpc.ServerUnaryCall<
+          discovery.UserFunctionError,
+          google_protobuf_empty_pb.Empty
+        >,
+        callback: grpc.sendUnaryData<google_protobuf_empty_pb.Empty>,
+      ) {
+        const msg = that.reportErrorLogic(
+          call.request.getCode(),
+          call.request.getMessage(),
+          call.request.getDetail(),
+          call.request.getSourceLocationsList(),
+        );
+
+        console.error(msg);
+        callback(null, new google_protobuf_empty_pb.Empty());
+      },
+      proxyTerminated(
+        call: grpc.ServerUnaryCall<
+          google_protobuf_empty_pb.Empty,
+          google_protobuf_empty_pb.Empty
+        >,
+        callback: grpc.sendUnaryData<google_protobuf_empty_pb.Empty>,
+      ) {
+        callback(null, new google_protobuf_empty_pb.Empty());
+      },
```

Comment:
This no longer runs the actual `proxyTerminated` logic now?

Reply:
good catch @pvlugter ! I'll double-check and fix it on top!